tactical-microgrid-standard/common/TimerHandler.h

```diff
@@ -185,7 +206,12 @@ class TimerHandler : public ACE_Event_Handler, protected TimerHolder<EventTypes>
   int handle_timeout(const ACE_Time_Value&, const void* arg)
   {
     Guard g(lock_);
-    auto timer = active_timers_[*reinterpret_cast<const TimerId*>(arg)];
+    auto timer_id = *reinterpret_cast<const TimerId*>(arg);
+    if (active_timers_.count(timer_id) == 0) {
+      ACE_ERROR((LM_WARNING, "(%P|%t) WARNING: TimerHandler::handle_timeout: timer id %d does NOT exist\n",
+                 timer_id));
```

Comment:
Is `TimerId` a type that can be logged with `%d` on all platforms?  It may need a cast or a different log formatter.

Reply:
Yes, `TimerId` is an alias for `long` so it can be logged with `%d`.

Reply:
ACE's logger will pass through `%d` to `snprintf` unchanged.  `%d` takes an `int` so on some systems it will read a 4-byte integer from the varargs but the `long` is 8 bytes.
https://en.cppreference.com/w/cpp/io/c/printf.html

Reply:
It looks like ACE's `%q` logger format will cover `long`.

Reply:
That's one option, but it doesn't match `long` on all platforms.  `%q` is always 8 bytes and `long` is sometimes 8 bytes.  So when using `%q` the parameter will need to be widened (or stay the same) to be always 8 bytes.